wavefront/server/apps/inference_app/inference_app/controllers/inference_controller.py

```
@@ -175,3 +46,30 @@ async def image_embedding(
         status_code=status.HTTP_200_OK,
         content=response_formatter.buildSuccessResponse(data={'response': embeddings}),
     )
+
+
+@inference_app_router.post('/v1/query/embeddings/batch')
+@inject
+async def image_embedding_batch(
+    payload: ImageBatchPayload,
+    response_formatter: ResponseFormatter = Depends(
+        Provide[CommonContainer.response_formatter]
+    ),
```

Ruff (0.15.7)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)